src/mono/mono/mini/driver.c

```diff
@@ -1909,24 +1909,29 @@ switch_gc (char* argv[], const char* target_gc)
 #endif
 }
 
-#ifdef TARGET_OSX
-
-/*
- * tries to increase the minimum number of files, if the number is below 1024
- */
 static void
-darwin_change_default_file_handles ()
+increase_descriptor_limit (void)
 {
+#if defined(HAVE_GETRLIMIT) && !defined(DONT_SET_RLIMIT_NOFILE)
 	struct rlimit limit;
 
-	if (getrlimit (RLIMIT_NOFILE, &limit) == 0){
-		if (limit.rlim_cur < 1024){
-			limit.rlim_cur = MAX(1024,limit.rlim_cur);
-			setrlimit (RLIMIT_NOFILE, &limit);
-		}
+	if (getrlimit (RLIMIT_NOFILE, &limit) == 0) {
+		// Set our soft limit for file descriptors to be the same
+		// as the max limit.
+		limit.rlim_cur = limit.rlim_max;
+#ifdef TARGET_OSX
```

Comment:
I was just copying whatever CoreCLR does here - but I guess the mobile platforms don't use CoreCLR?  If you want me to change anything here, please let me know - otherwise that might be done as a follow-on patch as well ...

Reply:
Yes mobile platforms use Mono right now. I was mainly wondering whether we should only do this on Desktop platforms but I guess it can't hurt to be consistent.

Let's change this to `__APPLE__` so it is consistent with CoreCLR's pal.cpp and will apply to iOS/tvOS as well.

```suggestion
#ifdef __APPLE__
```

Reply:
I just noticed that this is only called by `mono_main()` which the mobile platforms don't use (they do custom embedding) so ultimately it doesn't matter, but I'd still prefer the consistency with the CoreCLR code.

Reply:
Ah I see.  I've applied your change now.